### PR TITLE
List trees

### PR DESCRIPTION
**Pull request by ipovalyaev**
Server side API for quering list of trees and addition to test app to test this.

### CODE REVIEW

File: app/www/index.html [hunk 1]
```diff
@@ -10,7 +10,8 @@
             * Disables use of inline scripts in order to mitigate risk of XSS vulnerabilities. To change this:
                 * Enable inline JS: add 'unsafe-inline' to default-src
         -->
-        <meta http-equiv="Content-Security-Policy" content="default-src 'self' data: gap: https://ssl.gstatic.com 'unsafe-eval'; style-src 'self' 'unsafe-inline'; media-src *; img-src 'self' data: content:;">
```
Comment (bokova) [inline app/www/index.html]: why is this removed?

File: server/controllers/api/getlist.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	
+	app.post('/api/getlist', function (req, res) {
+		console.log('inside post');
+  		if (!req.body) return res.sendStatus(400);
+
+  		var requested = req.body;
+
+  		res.header("Access-Control-Allow-Origin", "*");
+
+		app.db.select([
```
Comment (bokova) [inline server/controllers/api/getlist.js:20]: If we return all of these, doesn't that make https://github.com/Learn-by-doing/hobobar/issues/6 obsolete?

File: app/www/index.html (hunk 1 above)
Comment (ipovalyaev) [reply app/www/index.html]: It's replaces for policy allowing  JSON to any server, until we decide about server name. Once we'll know about server name we can replace it with policy, allowing JSON to this particular server.

File: server/controllers/api/getlist.js (hunk 2 above)
Comment (ipovalyaev) [reply server/controllers/api/getlist.js:20]: Hmmmm... Idealy on requesting details of a tree we should return picture of it as well. I do suppose, sending full colunms from the table will not impact traffic too much, but we can discuss. Anyway as a part of #6 we should have a mechinism of getting picture in addition.

Comment (chill117) [reply server/controllers/api/getlist.js:20]: The picture data itself will be served statically from our server (normal HTTP behavior).

If the list trees end-point returns all the data of all the trees (including description, type, and image URL) then there is no need for the get detail API end-point.

File: server/package.json [hunk 3]
```diff
@@ -18,6 +18,7 @@
   "homepage": "https://github.com/Learn-by-doing/hobobar#readme",
   "dependencies": {
     "async": "2.1.5",
+    "body-parser": "^1.17.0",
```
Comment (chill117) [inline server/package.json:21]: Static version numbers for dependencies is preferred because in the future, the new person to the project could run `npm install` and get a different version from the rest of us. This could lead to strange bugs and confusion.

File: app/www/js/index.js [hunk 4]
```diff
@@ -105,9 +109,57 @@ document.addEventListener('deviceready', function() {    // The device is ready
...
+                        }
+                        else {
+                            alert("Got id:" + data[0].id + " description="+ data[0].description);
+                        }
+                    })
+                }
+        })
+
+
+    })(window);
```
Comment (chill117) [inline app/www/js/index.js]: I think this PR should only include the server part. There is a separate issue for implementing the get list of trees behavior in the mobile app.

File: server/controllers/api/getlist.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+module.exports = function(app) {
+	
+	var bodyParser = require('body-parser')
+	var jsonParser = bodyParser.json()
+	var urlencodedParser = bodyParser.urlencoded({ extended: false })
+
+	app.use(bodyParser.urlencoded({ extended: false }))
+	app.use(bodyParser.json())
+	
+	app.post('/api/getlist', function (req, res) {
```
Comment (chill117) [inline server/controllers/api/getlist.js:12]: The following is a good reference for what a REST API would look like:
https://stackoverflow.com/questions/671118/what-exactly-is-restful-programming

The HTTP verb and the URI are important because they tell anyone using the API how it works and what it does.

**Review by chill117 — CHANGES_REQUESTED** [review]

**Comment by ipovalyaev** [comment]
Replaces with a new one
